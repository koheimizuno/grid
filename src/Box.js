import React from 'react'
import styled from 'styled-components'
import { space, width, responsiveStyle } from 'styled-system'
import { number, string, array, oneOfType } from 'prop-types'
import tag from 'tag-hoc'
import propTypes from './propTypes'
import removeProps from './remove-props'

export const flex = responsiveStyle('flex')
export const order = responsiveStyle('order')

const Tag = tag(removeProps)
const Base = Tag('div')

const Box = styled(Base)([],
  { boxSizing: 'border-box' },
  width,
  space,
  flex,
  order
)
Box.displayName = 'Box'

const responsivePropType = oneOfType([
  number,
  string,
  array
])

Box.propTypes = Object.assign({}, propTypes, {
  flex: responsivePropType,
  order: responsivePropType
})

export default Box
