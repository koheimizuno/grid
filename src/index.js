export { default as Box } from './Box'
export { default as Flex } from './Flex'
export { default as theme } from './theme'

// for backwards compatibility
export { default as div } from 'clean-tag'
