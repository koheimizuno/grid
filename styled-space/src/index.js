import React from 'react'
import styled from 'styled-components'
import { space } from 'styled-system'

const classnames = (...args) => args.join(' ')
const getClassName = el => (el.props && el.props.className) || ''

export class StyledChildren extends React.Component {
  render () {
    const { className, children } = this.props
    const styledChildren = React.Children.toArray(children)
      .map(child => React.cloneElement(child, {
        className: classnames(getClassName(child), className)
      }))

    return (
      <React.Fragment>
        {styledChildren}
      </React.Fragment>
    )
  }
}

const Space = styled(StyledChildren)([], space)

Space.propTypes = space.propTypes

export default Space
