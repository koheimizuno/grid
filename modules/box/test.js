import test from 'ava'
import React from 'react'
import { create as render } from 'react-test-renderer'
import { space } from 'grid-styled-constants'
import Box, {
  parse,
  getProperties,
  px
} from './src/Box'

test('parses padding props', t => {
  const a = parse(space)({ p: 1 })
  const b = parse(space)({ pt: 2 })
  const c = parse(space)({ pr: 3 })
  const d = parse(space)({ pb: 4 })
  const e = parse(space)({ pl: 1 })
  const f = parse(space)({ px: 2 })
  const g = parse(space)({ py: 3 })
  t.is(a, 'padding:8px')
  t.is(b, 'padding-top:16px')
  t.is(c, 'padding-right:32px')
  t.is(d, 'padding-bottom:64px')
  t.is(e, 'padding-left:8px')
  t.is(f, 'padding-left:16px;padding-right:16px')
  t.is(g, 'padding-top:32px;padding-bottom:32px')
})

test('parses margin props', t => {
  const a = parse(space)({ m: 1 })
  const b = parse(space)({ mt: 2 })
  const c = parse(space)({ mr: 3 })
  const d = parse(space)({ mb: 4 })
  const e = parse(space)({ ml: 1 })
  const f = parse(space)({ mx: 2 })
  const g = parse(space)({ my: 3 })
  t.is(a, 'margin:8px')
  t.is(b, 'margin-top:16px')
  t.is(c, 'margin-right:32px')
  t.is(d, 'margin-bottom:64px')
  t.is(e, 'margin-left:8px')
  t.is(f, 'margin-left:16px;margin-right:16px')
  t.is(g, 'margin-top:32px;margin-bottom:32px')
})

test('renders', t => {
  const json = render(<Box m={2} px={3} />).toJSON()
  t.snapshot(json)
})
