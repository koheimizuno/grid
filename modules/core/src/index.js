import Box from 'box-styled'
import Flex from 'flex-styled'
import Grid from 'grid-styled-grid'

import Golden from './Golden'
import { Half, Third, Quarter } from './Fractions'

export { default as Box } from 'box-styled'
export { default as Flex } from 'flex-styled'
export { default as Grid } from 'grid-styled-grid'

export { default as Golden } from './Golden'
export { Half, Third, Quarter } from './Fractions'

export default {
  Box,
  Flex,
  Grid,
  Golden,
  Half,
  Third,
  Quarter
}
