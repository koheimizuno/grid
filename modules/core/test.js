import test from 'ava'
import GS from './src'

test('exports components', t => {
  t.is(typeof GS, 'object')
  t.is(typeof GS.Grid, 'function')
  t.is(typeof GS.Box, 'function')
  t.is(typeof GS.Flex, 'function')
  t.is(typeof GS.Half, 'function')
  t.is(typeof GS.Third, 'function')
  t.is(typeof GS.Quarter, 'function')
  t.is(typeof GS.Golden.A, 'function')
  t.is(typeof GS.Golden.B, 'function')
})
