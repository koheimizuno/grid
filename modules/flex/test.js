import React from 'react'
import test from 'ava'
import { create as render } from 'react-test-renderer'
import Flex from './src/Flex'

test('renders', t => {
  const flex = render(<Flex />)
  t.snapshot(flex)
})

