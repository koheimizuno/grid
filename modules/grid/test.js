import React from 'react'
import test from 'ava'
import { shallow } from 'enzyme'
import Grid, {
  Half,
  Third,
  Quarter,
  toArray,
  percent,
  media,
  width
} from './src/Grid'

test('toArray converts values to arrays', t => {
  const arr = [ 1, 2 ]
  const num = 2
  const a = toArray(arr)
  const b = toArray(num)
  t.deepEqual(a, [ 1, 2 ])
  t.deepEqual(b, [ 2 ])
})

test('percent converts numbers to percent values', t => {
  const a = percent(0)
  const b = percent(1/2)
  const c = percent(1)
  const d = percent(16)
  t.is(a, '0%')
  t.is(b, '50%')
  t.is(c, '100%')
  t.is(d, 16)
})

test('media wraps numbers in a media query string', t => {
  const a = media(40)
  t.is(a, '@media screen and (min-width: 40em)')
})

test.only('width converts width props to css', t => {
  const a = width([])({ width: 1 })
  const b = width([ null, 40 ])({ width: [ 1, 1/2 ] })
  t.is(a, 'width:100%;')
  console.log(b)
})

test('renders', t => {
  t.notThrows(() => {
    shallow(<Grid />)
  })
})

test('has a classname', t => {
  const wrapper = shallow(<Grid />)
  t.truthy(wrapper.props().className)
})

test.todo('generates a certain css string')

test('Half renders', t => {
  t.notThrows(() => {
    shallow(<Half />)
  })
})

test('Third renders', t => {
  t.notThrows(() => {
    shallow(<Third />)
  })
})

test('Quarter renders', t => {
  t.notThrows(() => {
    shallow(<Quarter />)
  })
})

