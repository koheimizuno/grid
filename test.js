import test from 'ava'
import React from 'react'
import { create as render } from 'react-test-renderer'
import {
  Box,
  Grid,
  Flex
} from './src'
import { flex } from './src/Box'

// Box
test('Box renders', t => {
  const json = render(<Box m={2} px={3} />).toJSON()
  t.snapshot(json)
})

test('Box renders with props', t => {
  const json = render(<Box
    m={[ 1, 2 ]}
    px={[ 1, 2 ]}
    w={1}
    flex='1 1 auto'
  />)
  t.snapshot(json)
})

test('Box renders with `is` prop', t => {
  const json = render(<Box is='section' />).toJSON()
  t.snapshot(json)
  t.is(json.type, 'section')
})

test('flex util returns null', t => {
  const sx = flex({})
  t.is(sx, null)
})

test('flex util returns a style object', t => {
  const sx = flex({ flex: 'none' })
  t.is(sx.flex, 'none')
})

// Grid
test('Grid renders', t => {
  const grid = render(<Grid />)
  t.snapshot(grid)
})

test('Grid has a classname', t => {
  const div = render(<Grid />).toJSON()
  t.truthy(div.props.className)
})

// Flex
test('Flex renders', t => {
  const flex = render(<Flex />)
  t.snapshot(flex)
})

test('Flex renders with props', t => {
  const flex = render(
    <Flex
      wrap
      direction='column'
      align='center'
      justify='space-between'
    />
  )
  t.snapshot(flex)
})

test('Flex renders with column prop', t => {
  const flex = render(
    <Flex
      column
    />
  )
  t.snapshot(flex)
})

test('Flex renders with responsive props', t => {
  const flex = render(
    <Flex
      wrap={[ true, false ]}
      direction={[ 'column', 'row' ]}
      align={[ 'stretch', 'center' ]}
      justify={[ 'space-between', 'center' ]}
    />
  )
  t.snapshot(flex)
})
