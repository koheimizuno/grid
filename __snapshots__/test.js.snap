// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders 1`] = `
<div
  className="sc-bdVaJa cahKXE"
  is={null}
/>
`;

exports[`Box renders with \`is\` prop 1`] = `
<section
  className="sc-bdVaJa iHZvIS"
  is={null}
/>
`;

exports[`Box renders with props 1`] = `
<div
  className="sc-bdVaJa juTkkz"
  is={null}
/>
`;

exports[`Flex renders 1`] = `
<div
  className="sc-htpNat eLQHN sc-bdVaJa iHZvIS"
  is={null}
/>
`;

exports[`Flex renders with column prop 1`] = `
<div
  className="sc-htpNat pyeRb sc-bdVaJa iHZvIS"
  is={null}
/>
`;

exports[`Flex renders with props 1`] = `
<div
  className="sc-htpNat deySwf sc-bdVaJa iHZvIS"
  is={null}
/>
`;

exports[`Flex renders with responsive props 1`] = `
<div
  className="sc-htpNat acOuK sc-bdVaJa iHZvIS"
  is={null}
/>
`;

exports[`Grid renders 1`] = `
<div
  className="sc-bwzfXH fIMJGd sc-bdVaJa iHZvIS"
  is={null}
/>
`;
